tests/middleware/test_message_state.py

```diff
@@ -0,0 +1,134 @@
+    ):
+        @remoulade.actor
+        def wait():
+            time.sleep(0.3)
+
+        stub_broker.declare_actor(wait)
+        msg = wait.send()
+        # We wait the message be emited
+        time.sleep(0.1)
+        assert (
+            state_backend.get_state(msg.message_id)["name"] == CatalogStates.Started.name
+        )
+
+    def test_failure_state_message(
+        self, stub_broker, stub_worker, state_middleware, state_backend
+    ):
+        @remoulade.actor
+        def error():
+            raise Exception()
+
+        remoulade.declare_actors([error])
+        msg = error.send()
+        stub_broker.join(error.queue_name)
+        stub_worker.join()
+        assert (
+            state_backend.get_state(msg.message_id)["name"] == CatalogStates.Failure.name
+        )
+
+    def test_cancel_state_message(
+        self,
+        stub_broker,
+        stub_worker,
+        state_backend,
+        cancel_backend,
+        state_middleware,
+        do_work,
+    ):
+        stub_broker.add_middleware(Cancel(backend=cancel_backend))
+        #  Pause the worker to be able to cancel the  message, and
+        #   this does not been processed  when it enters the queue
+        stub_worker.pause()
+        msg = do_work.send(5)
+        msg.cancel()
+        stub_worker.resume()
+        stub_broker.join(do_work.queue_name)
+        stub_worker.join()
+        assert (
+            state_backend.get_state(msg.message_id)["name"] == CatalogStates.Canceled.name
+        )
+
+    def test_skip_state_message(
+        self, stub_broker, stub_worker, state_backend, state_middleware, do_work
+    ):
+        class SkipMiddleware(Middleware):
+            def before_process_message(self, broker, message):
+                raise SkipMessage()
+
+        stub_broker.add_middleware(SkipMiddleware())
+        msg = do_work.send(5)
+        stub_broker.join(do_work.queue_name)
+        stub_worker.join()
+        assert (
+            state_backend.get_state(msg.message_id)["name"] == CatalogStates.Skipped.name
+        )
+
+    @pytest.mark.parametrize(
+        "ttl, state, result_type",
+        [
+            pytest.param(1000, CatalogStates.Failure.name, {}),
+            pytest.param(0, CatalogStates.Success.name, None),
+        ],
+    )
+    def test_expiration_data_backend(
+        self, ttl, state, result_type, stub_broker, state_backend
+    ):
+        @remoulade.actor
+        def wait():
+            time.sleep(0.3)
+
+        stub_broker.add_middleware(MessageState(backend=state_backend, state_ttl=ttl))
+        stub_broker.declare_actor(wait)
+        msg = wait.send()
+        time.sleep(0.1)
+        data = state_backend.get_state(msg.message_id)
+        # if the ttl is  greater than the expiration, the data should be None
+        assert type(data) == type(result_type)
+
+    @pytest.mark.parametrize(
+        "msize,x,expected", [pytest.param(0, 5, []), pytest.param(1000, 5, [5])],
```

Comment:
use full name when possible: `max_size`, actually x is not changing so no need to parametrize it.